llvm/lib/Target/Hexagon/HexagonTargetTransformInfo.cpp

```diff
@@ -88,9 +88,8 @@ void HexagonTTIImpl::getPeelingPreferences(Loop *L, ScalarEvolution &SE,
                                            TTI::PeelingPreferences &PP) {
   BaseT::getPeelingPreferences(L, SE, PP);
   // Only try to peel innermost loops with small runtime trip counts.
-  if (L && L->isInnermost() && canPeel(L) &&
-      SE.getSmallConstantTripCount(L) == 0 &&
-      SE.getSmallConstantMaxTripCount(L) > 0 &&
+  if (L && L->isInnermost() && canPeel(L) && !SE.getSmallConstantTripCount(L) &&
+      SE.getSmallConstantMaxTripCount(L) &&
       SE.getSmallConstantMaxTripCount(L) <= 5) {
```

Comment:
```suggestion
      SE.getSmallConstantMaxTripCount(L) <= 5) {
```
I think that should work now? Calling the method twice is not great...

Reply:
`std::optional<T>::operator<=(const optional<T>& opt, const T& value)` returns true when `opt` is `std::nullopt`, so this won't work.